ci/tests/api-functionality/api_test.sh

```diff
@@ -3,6 +3,6 @@
 set -exo pipefail
 
 #Just print the response for verbosity before testing the output
-curl -s -XGET -H "Accept: application/json" "http://localhost:8080/smoke-test-api/?arg=test"
-curl -s -XGET -H "Accept: application/json" "http://localhost:8080/smoke-test-api/?arg=test"| jq -e '.args.arg == "test"'
+curl -s --retry 5 --retry-delay 10 -XGET -H "Accept: application/json" "http://localhost:8080/smoke-test-api/?arg=test"
```

Comment:
Needs to retry when connection failed (which is what happens if the service is not up. `--retry-connfailed` does the job.

Reply:
It seems like there's no `--retry-connfailed`, but `--retry-connrefused`. That should work as well, since the port is not bound till service is up, and will return connrefused.

Reply:
Sorry, yes that is what I meant.